backend/dataall/modules/datapipelines/db/repositories.py

```diff
@@ -0,0 +1,156 @@
+from sqlalchemy import or_, and_
+from sqlalchemy.orm import Query
+
+from dataall.core.group.services.group_resource_manager import GroupResource
+from dataall.db import models, exceptions, paginate
+from dataall.modules.datapipelines.db.models import DataPipeline, DataPipelineEnvironment
+
+
+class DatapipelinesRepository(GroupResource):
+    """DAO layer for datapipelines"""
+    _DEFAULT_PAGE = 1
+    _DEFAULT_PAGE_SIZE = 10
+
+    @staticmethod
+    def count_resources(session, environment, group_uri) -> int:
```

Comment:
This method should not be static. Otherwise, python will try to pass `self` to it which can lead to a bug